packages/hawtio/src/plugins/camel/routes/RouteDiagram.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useCallback, useContext, useEffect, useState } from 'react'
+import { CamelContext } from '@hawtiosrc/plugins/camel/context'
+import ReactFlow, {
+  addEdge,
+  ConnectionLineType,
+  Edge,
+  Handle,
+  Node,
+  NodeProps,
+  NodeToolbar,
+  Position,
+  useEdgesState,
+  useNodesState,
+} from 'reactflow'
+import dagre from 'dagre'
+import { CamelNodeData, routeVizualisationService } from '@hawtiosrc/plugins/camel/routes/visualization-service'
+import 'reactflow/dist/style.css'
+import './routeDiagram.css'
+import { Flex, FlexItem, Page } from '@patternfly/react-core'
+
+const dagreGraph = new dagre.graphlib.Graph()
+dagreGraph.setDefaultEdgeLabel(() => ({}))
+
+const nodeWidth = 300
+const nodeHeight = 86
+
+export type ICamelNode = Node & {
+  data: CamelNodeData
+}
+
+const getLayoutedElements = (nodes: ICamelNode[], edges: Edge[], direction = 'TB') => {
```

Comment:
Maybe you can put those dagre specific code, i.e. this function and const `dagreGraph`, into another file or move them into `visualizationService` if that makes sense, in order to keep the React components simple.